modules/images/webp-uploads/helper.php

```diff
@@ -84,19 +83,23 @@ function webp_uploads_generate_additional_image_source( $attachment_id, $image_s
 	 * @return array|null|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
 	 */
 	$image = apply_filters( 'webp_uploads_pre_generate_additional_image_source', null, $attachment_id, $image_size, $size_data, $mime );
-
 	if ( is_wp_error( $image ) ) {
 		return $image;
 	}
 
 	if (
-		is_array( $image )
-		&& ! empty( $image['file'] )
-		&& ! empty( $image['path'] )
+		is_array( $image ) &&
+		! empty( $image['file'] ) &&
+		(
+			! empty( $image['path'] ) ||
+			array_key_exists( 'filesize', $image )
+		)
 	) {
 		return array(
 			'file'     => $image['file'],
-			'filesize' => filesize( $image['path'] ),
+			'filesize' => array_key_exists( 'filesize', $image )
+				? $image['filesize']
+				: filesize( $image['path'] ),
```

Comment:
We are also returning same array in later in this function. https://github.com/WordPress/performance/pull/334/files#diff-1f653bd37f60a6476485a6794f7003a91b3713868c7414588d5835332fe68642R156

so can we put checks there as well?